raiden/api/rest.py

```diff
@@ -1244,13 +1296,22 @@ def patch_channel(
                 status_code=HTTPStatus.CONFLICT,
             )
 
-        if total_deposit is None and state is None and total_withdraw is None:
+        if reveal_timeout is not None and total_deposit is not None:
             return api_error(
                 errors=(
-                    "Nothing to do. Should either provide "
-                    "'total_deposit', 'total_withdraw' or 'state' argument"
+                    "Can not update a channel's reveal timeout "
+                    "and total deposit at the same time"
                 ),
-                status_code=HTTPStatus.BAD_REQUEST,
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+        if reveal_timeout is not None and total_withdraw is not None:
```

Comment:
Some of these are too strict, I believe it is perfectly fine to do a reveal timeout update and a withdraw and deposit at the same time.

This can be improved if necessary in the future.